pkg/mtls/singleton.go

```diff
@@ -0,0 +1,66 @@
+package mtls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"github.com/stackrox/rox/pkg/buildinfo"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	once     sync.Once
+	instance *certificateParserWrapper
+	log      = logging.LoggerForModule()
+)
+
+// certificateParserWrapper is a singleton that wraps the CertificateParser interface.
+// This allows us to override the implementation of the CertificateParser interface for testing purposes.
+// This override mechanism is protected in ReleaseBuilds, and it will only work for DevelopmentBuilds.
+type certificateParserWrapper struct {
+	parser CertificateParser
+}
+
+// Override the internal parser. This should only be used for testing.
+func (c *certificateParserWrapper) Override(parser CertificateParser) {
+	if buildinfo.ReleaseBuild {
+		// This is not allowed in release builds
+		log.Errorf("The CertificateParser should not be override in production")
```

Comment:
```suggestion
		log.Errorf("The CertificateParser should not be overridden in production")
```

Reply:
I’ve separated the implementation of `Override` into two files (_dev and _release) so there’s no need for this check.